### PR TITLE
Add hostlookup tests

### LINKED ISSUE #57: [IDEA] FHOSTLOOKUP: Automated test & Hardened
**What is your feature about ?**
- Adding automated test for the core component
- Valgrind test for Linux
- Correcting any issue that occurs during testing

**Additional context**
Automated testing with valgrind would be nice (look for XML output).

### PR DESCRIPTION
**Pull request by NS4nti**
# :sparkles: Add test for HOSTLOOKUP filter

**CAUTION**: Merge after #70, #72

:bangbang: Once all the **checklist** is **done** you have to:
  * **stash merge** this pull request 
  * **delete** the corresponding **branch** 
  * **close** the associated **issue**
  * **edit** : https://github.com/VultureProject/darwin/wiki/Filters-Summary to add fhostlookup filter to *Tested and approved* section.
## :black_nib: Description

Add scenarii test for the filter [hostlookup](https://github.com/VultureProject/darwin/wiki/Filter-HostLookup).

**Tested under:**
- **FreeBSD 12.0**
  - Redis 4.0.14
  - Boost 1.70.0
  - clang++ 6.0.1
  - CMake 3.14.5
  - Python 3.6.9
- **Ubuntu 18.04**
  - Redis 4.0.9
  - Boost 1.70.0
  - g++ 8.3.0
  - CMake 3.10.2
  - Python 3.6.8
  - Valgrind 3.13.0

**Test list:**
- Untrusted host received
- Trusted host received
- Multiple untrusted host received
- Multiple trusted host received
- Multiple host received

Resolve #57 

## :page_with_curl: Type of change
- [x] New feature (non-breaking change which adds functionality)

## :heavy_check_mark: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

</br>

- [x] :raising_hand: **I certify on my honor that all the informations provided are true, and I've done all I can to deliver a high quality code**

### CODE REVIEW

**Comment by HugoSoszynski** [comment]
Could you please add tests for the hosts "init_data" file as mentioned in comment of #57 ?

It may be a good idea to document the behavior of the filter against the initialization file in different cases _(blank line, \r\n, spaces and tabs, etc.)_ to avoid surprises :)

**Comment by NS4nti** [comment]
> Could you please add tests for the hosts "init_data" file as mentioned in comment of #57 ?
> 
> It may be a good idea to document the behavior of the filter against the initialization file in different cases _(blank line, \r\n, spaces and tabs, etc.)_ to avoid surprises :)

Done with a function `GetLineSafe` from a Files.cpp (from [here](https://gist.github.com/josephwb/df09e3a71679461fc104)) and tests added

**Comment by frikilax** [comment]
Why is this PR referenced to #73 ?

File: tests/filters/fhostlookup.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        unique_untrusted_host_test,
+        unique_trusted_host_test,
+        multiple_trusted_host_test,
+        multiple_untrusted_host_test,
+        multiple_host_test,
+        init_data_test
+    ]
+
+    for i in tests:
+        print_result("connection: " + i.__name__, i())
```
Comment (frikilax) [inline tests/filters/fhostlookup.py]: connection -> hostlookup ?

File: tests/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+git+https://github.com/VultureProject/darwin-client-python.git
```
Comment (frikilax) [inline tests/requirements.txt]: yep

File: tests/filters/fhostlookup.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        filter_code="HOSTLOOKUP",
+        response_type="back",
+    )
+
+    # VERIFY RESULTS
+    certitudes = results.get('certitude_list')
+
+    if certitudes is None:
+        ret = False
+        logging.error("Hostlookup Test : {} : No certitude list found in result".forma(test_name))
```
Comment (frikilax) [inline tests/filters/fhostlookup.py]: you sure about the "forma" here ?

Comment (NS4nti) [reply tests/filters/fhostlookup.py]: done !

File: tests/filters/fhostlookup.py (hunk 1 above)
Comment (NS4nti) [reply tests/filters/fhostlookup.py]: done

**Pull request merged**